### PR TITLE
Periodic reporting params

### PR DESCRIPTION
**Pull request by gaborgsomogyi**
Periodic reporting params added

### CODE REVIEW

File: app/src/main/res/layout/fragment_send_command.xml [hunk 1]
```diff
@@ -14,18 +14,79 @@
...
+                <TextView
+                    android:id="@+id/frequencyText"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:text="@string/command_frequency_frequency"
+                    android:layout_gravity="center"
+                    android:textAppearance="?android:attr/textAppearanceLarge" />
+
+                    <NumberPicker
```
Comment (tananaev) [inline app/src/main/res/layout/fragment_send_command.xml]: wrong indentation

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -29,6 +33,9 @@
     <string name="action_show_on_map">Show on map</string>
     <string name="action_send_command">Send command</string>
 
+    <string name="command_command">Command:</string>
```
Comment (tananaev) [inline app/src/main/res/values/strings.xml]: I think we shouldn't have colon symbols

File: app/src/main/java/org/traccar/manager/SendCommandFragment.java [hunk 3]
```diff
@@ -146,4 +167,17 @@ public void onSuccess(Response<Command> response) {
         return view;
     }
 
+    private String getI10NString(String key) {
```
Comment (tananaev) [inline app/src/main/java/org/traccar/manager/SendCommandFragment.java]: Why is "N" capital?

File: app/src/main/java/org/traccar/manager/SendCommandFragment.java [hunk 4]
```diff
@@ -146,4 +167,17 @@ public void onSuccess(Response<Command> response) {
...
+        String result = key;
+
+        Integer resId = i10nMapping.get(key);
+        if(resId != null) {
+            try {
+                CharSequence nameCharSequence = getContext().getResources().getText(resId);
+                result = nameCharSequence.toString();
+            } catch (Resources.NotFoundException e) {}
```
Comment (tananaev) [inline app/src/main/java/org/traccar/manager/SendCommandFragment.java]: Need at least log something if there is an exception

File: app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    private LinearLayout unitGroup;
+    private Spinner unitSpinner;
+
+    public PositionPeriodicCommandHandler(View view, Resources resources) {
+        frequencyGroup = (LinearLayout) view.findViewById(R.id.frequencyGroup);
+        frequencyNumberPicker = (NumberPicker) view.findViewById(R.id.frequency);
+        unitGroup = (LinearLayout) view.findViewById(R.id.unitGroup);
+        unitSpinner = (Spinner) view.findViewById(R.id.unit);
+
+        frequencyNumberPicker.setMinValue(0);
```
Comment (tananaev) [inline app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java]: Why do we need limits and why are they hardcoded?

**Comment by tananaev** [comment]
What's the purpose of `CommandHandler`? I think it complicates things.

I've put some comments in the diff as well.

File: app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java (hunk 5 above)
Comment (gaborgsomogyi) [inline app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java]: The default limit is [0, 0] so something must be set. The question is where and what. When I've tries Integer.MAX_VALUE it was ugly. Suggestions?

**Comment by gaborgsomogyi** [comment]
Related the CommandHandler. If there is interface or base class the handlers can be handled in a general way (in a list this case). If a new handler is implemented in the future only one additional line is necessary in the SendCommandFragment. Something like this:

commandHandlers.add(new NewCustomCommandHandler());

File: app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java (hunk 5 above)
Comment (tananaev) [inline app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java]: Is there a way to type the value with NumberPicker?

**Comment by tananaev** [comment]
I still don't like the idea of `CommandHandler`. First of all, the name is confusing. The class basically updates the UI. Also, instead of having some mapping, you go through all the command handlers and call them all. Method names are weird as well.

Let's go through each method:

- `constructor` - doesn't do anything useful
- `onCommandSelected` - this is the main method, but I think it doesn't have enough complexity to have a separate class hierarchy to handle it
- `onCommandNothingSelected` - duplicates `onCommandSelected`
- `onCommandAddParameters` - this is very simple method as well

You still have widgets in the fragment, so you are not actually decoupling fragment from the command types. I think for now we can keep all the logic in the fragment (this is also consistent with web app).

File: app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java (hunk 5 above)
Comment (gaborgsomogyi) [inline app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java]: Basically yes, but only when one sets the min/max value. With the mentioned default settings only the number 0 is accepted.

Comment (tananaev) [inline app/src/main/java/org/traccar/manager/commandhandler/PositionPeriodicCommandHandler.java]: I can suggest to use text view with numeric input. I doubt anybody would be interested to swipe through seconds anyway.

**Comment by gaborgsomogyi** [comment]
Changes are done.

**Comment by tananaev** [comment]
Merged. Thanks for pull request.

**Pull request merged**